Protocol-Independent Cross-Origin Discovery Flow

hlflanagan opened the pull request:
Attempting to capture the multilateral federation use case, specifically around WAYF services.

hlflanagan committed:
Create MultilateralFederation.md

gffletch commented on src/scenarios/MultilateralFederation.md:
```diff
@@ -0,0 +1,60 @@
+- Reference: <https://www.oasis-open.org/committees/download.php/35389/sstc-saml-profiles-errata-2.0-wd-06-diff.pdf>, <https://kantarainitiative.github.io/SAMLprofiles/fedinterop.html>, <https://kantarainitiative.github.io/SAMLprofiles/saml2int.html>
+
+#### Browser Features Required
+- 1st party Cookie
+- 3rd party cookies
+- Redirect with link decoration
+- Persistent Local Storage
+- cross-domain postMessage
```
Would it be possible to highlight in the 'Description of flow' section how each of these are used? The use of local storage is already well described. I think it would be helpful for the scenario to highlight how 3rd party cookies are required as well as cross-domain postMessage.

hlflanagan replied:
I think I went overboard with the pieces that are required for this specific use case. I've updated the file and updated the drawings to try and be more clear about exactly what/where the different components interact with browser features.

hlflanagan committed:
Update MultilateralFederation.md

Updated required features, updated link to diagram

hlflanagan committed:
significant revisions

removing any protocol dependencies; added a better sequence diagram link to the description of the flow

gffletch commented on src/scenarios/cross-origin-discovery.md:
```diff
@@ -0,0 +1,57 @@
+### Description Of The Flow
+A full design of the data flows used to support a central discovery service are available here: <https://docs.google.com/presentation/d/1h3XQ6BtTQ7KJkBQkIkWjktp3RXqHDXEKsa5ymrjzA-k/edit#slide=id.p> 
+
+This flow writes to browser local storage via a dedicated component which loads a non-displayed iFrame.
```
It would be great if there could be a bit more detail around how the iframe interacts with the local storage and where the iframe is embedded. 

Less important: I'd prefer the seq diag to be included (I just added one to my scenario using the Online Server of https://plantuml.com/).
